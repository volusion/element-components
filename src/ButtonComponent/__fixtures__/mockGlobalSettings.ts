import { GlobalSettings } from '../types';
export const mockGlobalSettings: GlobalSettings = {
    globalComponents: {
        volComponentButton: {
            primaryButtonStyles: {
                size: 'large',
                rounded: 'none',
                textColor: '#fff',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: true,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: '#fff',
                backgroundColor: 'rgba(247, 68, 186, 1)',
                borderThickness: 'none',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(218, 15, 148, 1)'
            },
            secondaryButtonStyles: {
                size: 'block',
                rounded: 'none',
                textColor: 'rgba(255, 255, 255, 1)',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: false,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: 'rgba(255, 255, 255, 1)',
                backgroundColor: 'rgba(247, 74, 201, 1)',
                borderThickness: 'thin',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(0, 0, 0, 1)'
            }
        }
    }
};
