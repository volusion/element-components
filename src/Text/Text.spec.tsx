import React from 'react';
import { mount } from 'enzyme';
import { block as Text, defaultConfig } from './index';
import { mockGlobalSettings as globalSettings } from '../__fixtures__/mockGlobalSettings';

describe('The <Text /> Component', () => {
    const renderWithContent = (content: string) => mount(<Text htmlString={content} globalSettings={globalSettings} />).render()

    describe('with default props', () => {
        it('matches the snapshot', () => {
            const rendered = renderWithContent(defaultConfig.htmlString);
            expect(rendered).toMatchSnapshot();
        });
    });

    describe('with empty content', () => {
        it('matches the snapshot', () => {
            expect(renderWithContent('')).toMatchSnapshot();
        });
    })

    describe('with custom content', () => {
        it('matches the snapshot', () => {
            expect(renderWithContent('<h1>Test Content</h1>')).toMatchSnapshot();
            expect(renderWithContent('Test Content')).toMatchSnapshot();
            expect(renderWithContent('<p>Test Content</p>')).toMatchSnapshot();
        });
    })
});

