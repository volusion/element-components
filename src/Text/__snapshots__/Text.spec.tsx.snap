// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Text /> Component with custom content matches the snapshot 1`] = `
<div>
  <h1>
    Test Content
  </h1>
</div>
`;

exports[`The <Text /> Component with custom content matches the snapshot 2`] = `
<div>
  Test Content
</div>
`;

exports[`The <Text /> Component with custom content matches the snapshot 3`] = `
<div>
  <p>
    Test Content
  </p>
</div>
`;

exports[`The <Text /> Component with default props matches the snapshot 1`] = `
<div>
  Lorem Ipsum
</div>
`;

exports[`The <Text /> Component with empty content matches the snapshot 1`] = `null`;
